typescript/packages/common-runner/src/cell.ts

```diff
@@ -145,14 +153,26 @@ export function cell<T>(value?: T): CellImpl<T> {
     get value(): T {
       return value as T;
     },
-    get entityId(): string | undefined {
+    get entityId(): EntityId | undefined {
       return entityId;
     },
-    set entityId(id: string) {
+    set entityId(id: EntityId) {
       if (entityId) throw new Error("Entity ID already set");
       entityId = id;
       setCellByEntityId(id, self);
     },
+    generateEntityId: (cause?: any): void => {
+      entityId = createRef(
+        typeof value === "object" && value !== null
+          ? (value as Object)
+          : value !== undefined
+          ? { value }
+          : {},
+        cause
+      );
+      setCellByEntityId(entityId, self);
+    },
+    toJSON: () => entityId?.toJSON(),
```

Comment:
while it is covered in tests that it does this - it seems like actual usage (like window-manager) doesn't leverage this fact?)

Reply:
Good question, I should add a comment.

The idea is that .toJSON is called when serializing to storage, and that if a cell appears inside another structure, that it should be encoded as a link to that cell. We should also make sure that cell is persisted. Then when loading, we'll encounter that link, and know to query for that cell by id. At first this might be actually separate queries, and eventually this will all happen at the synopsis layer.